sdk/core/core-lro/samples-dev/customPoller.ts

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @summary demnostrates how to create a custom poller.
+ */
+
+import type { OperationState, OperationStatus, PollerLike } from "@azure/core-lro";
+import { delay } from "@azure/core-util";
+
+const DEFAULT_POLL_INTERVAL_IN_MS = 2000;
+
+type OperationConfig = { id: string };
+
+type RestorableOperationState<TResult, TState extends OperationState<TResult>> = TState & {
+  config: OperationConfig;
+};
+
+async function initOperation<T>(): Promise<RestorableOperationState<T, OperationState<T>>> {
+  // starts the operation
+  return { status: "running", config: { id: "1" } };
+}
+
+async function pollOperation<T>({
+  setDelay,
+  state,
+  options,
+}: {
+  state: RestorableOperationState<T, OperationState<T>>;
```

Comment:
Should we move this into its own type? Just to keep the sample a little clearer

Reply:
Addressed in https://github.com/Azure/azure-sdk-for-js/pull/32469/commits/9babca3eabce977777fdbb2aba1959735d48fcd2